crates/rpc-types-beacon/src/requests.rs

```diff
@@ -0,0 +1,181 @@
+                        WITHDRAWAL_REQUEST_TYPE => {
+                            self.withdrawals = Self::parse_request_payload(
+                                payload,
+                                MAX_WITHDRAWAL_REQUESTS_PER_BLOCK,
+                                WITHDRAWAL_REQUEST_TYPE,
+                            )?;
+                        }
+                        CONSOLIDATION_REQUEST_TYPE => {
+                            self.consolidations = Self::parse_request_payload(
+                                payload,
+                                MAX_CONSOLIDATION_REQUESTS_PER_BLOCK,
+                                CONSOLIDATION_REQUEST_TYPE,
+                            )?;
+                        }
+                        unknown => return Err(TryFromRequestsError::UnknownRequestType(unknown)),
+                    }
+
+                    Ok(self)
+                }
+            }
+
+            let accumulator = value
+                .iter()
+                .try_fold(RequestAccumulator::default(), |acc, request| acc.accumulate(request))?;
+
+            Ok(Self {
+                deposits: accumulator.deposits,
+                withdrawals: accumulator.withdrawals,
+                consolidations: accumulator.consolidations,
+            })
+        }
+    }
+
+    /// Errors possible converting a [Requests] to [ExecutionRequestsV4]
+    #[derive(Debug, thiserror::Error)]
+    pub enum TryFromRequestsError {
+        /// One of the Bytes is empty.
+        #[error("empty bytes in requests body")]
+        EmptyRequest,
+        /// Bytes prefix is not a known EIP-7685 request_type in Electra.
+        #[error("unknown request_type prefix: {0}")]
+        UnknownRequestType(u8),
+        /// Remaining bytes could not be decoded as SSZ requests_data.
+        #[error("ssz error decoding requests_type: {0}")]
+        SszDecodeError(u8, DecodeError),
+        /// Requests of request_type exceeds Electra size limits
+        #[error("requests_data payload for request_type {0} exceeds Electra size limit {1}")]
+        RequestPayloadSizeExceeded(u8, usize),
+    }
+
+    impl From<&ExecutionRequestsV4> for Requests {
+        fn from(val: &ExecutionRequestsV4) -> Self {
+            let deposit_bytes = val.deposits.as_ssz_bytes();
+            let withdrawals_bytes = val.withdrawals.as_ssz_bytes();
+            let consolidations_bytes = val.consolidations.as_ssz_bytes();
+
+            let mut requests = Self::with_capacity(3);
+            requests.push_request_with_type(DEPOSIT_REQUEST_TYPE, deposit_bytes);
+            requests.push_request_with_type(WITHDRAWAL_REQUEST_TYPE, withdrawals_bytes);
+            requests.push_request_with_type(CONSOLIDATION_REQUEST_TYPE, consolidations_bytes);
+            requests
+        }
+    }
+
+    impl ExecutionRequestsV4 {
+        /// Convert the [ExecutionRequestsV4] into a [Requests].
+        pub fn to_requests(&self) -> Requests {
+            self.into()
+        }
+    }
```

Comment:
can we move this top of mod?

Reply:
Yup, moved in https://github.com/alloy-rs/alloy/pull/1787/commits/daf77039c827bb5491f570fb7b64158339b45682.  Required some addiitonal `#[cfg(feature = "ssz")` pragmas though.